### ISSUE TITLE
Allow non-supported languages in style elements to be skipped

### ISSUE THREAD

**Issue by chrisvfritz** [stylelint/stylelint#1686]
This is a feature request and possibly a new option.

## The problem

When processing HTML, there are some applications where `style` elements may contain code that will later be compiled to CSS. The problem is that there's currently no way to tell stylelint that a `style` element contains an unsupported language and should be skipped.

### Automatically detect non-CSS style tags

In cases mentioned above, the language inside a style tag is typically detected via the `type` or `lang` attributes. For example:

``` html
<style type="text/scss">
<!-- OR -->
<style lang="scss">
```

Perhaps [here](https://github.com/stylelint/stylelint/blob/0e5048872babf8d1f9bb73e7ec0634984b037d82/src/__tests__/fixtures/processor-html.js#L59-L61), the `style` element could be skipped when either the `type` and/or `lang` attributes exist __and__ they do not contain a valid language.

### Allow specific `style` elements to be ignored

When scanning a project directory, we can choose to exclude specific files in an `ignoreFiles` option. Similarly, it would be nice if it were possible to filter out specific `style` elements, possibly  using the `querySelector` API. For example:

``` json
"ignoreStyleElements": [
  "[lang*='styl']",
  "[lang='postcss']"
]
```

**Comment by Arcanemagus** [comment]
Just as an FYI, that entire functionality was removed in v7.0.0 (https://github.com/stylelint/stylelint/pull/1535). Nobody has taken up the mantle of writing the HTML parser as of yet.

**Comment by davidtheclark** [comment]
Yep, so I'm going to close this issue as it needs to be addressed in the processor that someone will hopefully build. (Whatever functionality was in place before v7 will not be improved upon in stylelint core.)

**Comment by jeddy3** [comment]
@chrisvfritz FYI, there's [a section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/processors.md) in the Developer Guide on writing processors. And [a section](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#processors) within the User Guide on their usage.
